test/e2e/ginkgo/multiple_cluster_deployment_test.go

```diff
@@ -0,0 +1,305 @@
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
+
+		// 1. Create object 1 with label A and label B, object 2 with label B,
+		// and a Placement that matches labels A AND B, and verify that only
+		// object 1 matches and is delivered.
+		// 2. Patch the cluster selector and expect objects to be removed and downsynced correctly.
+		ginkgo.It("downsync objects that fully match on object and cluster selector", func() {
+			ginkgo.By("create two deployments and a bindingpolicy that matches only one")
+			util.DeleteDeployment(ctx, wds, ns, "nginx")
+			util.CreateDeployment(ctx, wds, ns, "one",
+				map[string]string{
+					"label1": "test1",
+				})
+			util.CreateDeployment(ctx, wds, ns, "two",
+				map[string]string{
+					"label1": "test1",
+					"label2": "test2",
+				})
+			util.CreateBindingPolicy(ctx, ksWds, "both-labels",
+				[]metav1.LabelSelector{
+					{MatchLabels: map[string]string{"name": "cluster1"}},
+				},
+				[]ksapi.DownsyncObjectTest{
+					{ObjectSelectors: []metav1.LabelSelector{
+						{MatchLabels: map[string]string{
+							"label1": "test1",
+							"label2": "test2",
+						}},
+					}}})
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+
+			ginkgo.By("patch the cluster selector")
+			patch := []byte(`{"spec": {"clusterSelectors": [{"matchLabels": {"name": "cluster2"}}]}}`)
+			_, err := ksWds.ControlV1alpha1().BindingPolicies().Patch(
+				ctx, "both-labels", types.MergePatchType, patch, metav1.PatchOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+		})
+
+		/***
+		// Broken until PR 1884 is merged.
+		ginkgo.It("handles OR of cluster and object selectors", func() {
+			util.DeleteDeployment(ctx, wds, ns, "nginx")
+			util.CreateDeployment(ctx, wds, ns, "one",
+				map[string]string{
+					"label1": "A",
+				})
+			util.CreateDeployment(ctx, wds, ns, "two",
+				map[string]string{
+					"label1": "B",
+				})
+			util.CreateDeployment(ctx, wds, ns, "three",
+				map[string]string{
+					"label1": "C",
+				})
+			util.CreateBindingPolicy(ctx, ksWds, "two-cluster-selectors",
+				[]metav1.LabelSelector{
+					{MatchLabels: map[string]string{"name": "cluster1"}},
+					{MatchExpressions: []metav1.LabelSelectorRequirement{
+						{Key: "name", Operator: metav1.LabelSelectorOpIn, Values: []string{"cluster2"}},
+					}},
+				},
+				[]ksapi.DownsyncObjectTest{
+					{ObjectSelectors: []metav1.LabelSelector{
+						{MatchLabels: map[string]string{"label1": "A"}},
+						{MatchLabels: map[string]string{"label1": "B"}},
+					}}})
+			util.ValidateNumDeployments(ctx, wec1, ns, 2)
+			util.ValidateNumDeployments(ctx, wec2, ns, 2)
+		})
+		***/
+
+		ginkgo.It("downsync based on label and object name", func() {
```

Comment:
label on what? DownsyncObjectTest can test labels on objects and also can test labels on the objet's Namespace.

Reply:
changed to object labels.